contrib/platform/src/com/sun/jna/platform/unix/LibCAPI.java

```diff
@@ -223,18 +218,155 @@ public ssize_t(long value) {
+     */
+    int fexecve(int fd, String[] argv, String[] env);
+
+    int WAIT_ANY = -1;
+    int WAIT_MYPGRP = 0;
+
+    /**
+     * The waitpid function is used to request status information from a child process.
+     * Normally, the calling process is suspended until the child process makes status information available by terminating.
+     * <p>
+     * If status information for a child process is available immediately, this function returns immediately without waiting.
+     * If more than one eligible child process has status information available, one of them is chosen randomly, and its status is returned immediately.
+     * To get the status from the other eligible child processes, you need to call waitpid again.
+     * <p>
+     * This function is a cancellation point in multi-threaded programs.
+     * This is a problem if the thread allocates some resources (like memory, file descriptors, semaphores or whatever) at the time waitpid is called.
+     * If the thread gets canceled these resources stay allocated until the program ends.
+     * To avoid this calls to waitpid should be protected using cancellation handlers.
+     * @param pid       the process ID of the child process to request information from.
+     *                  Can also be {@link #WAIT_ANY}, which requests status information for any child process.
+     *                  Can also be {@link #WAIT_MYPGRP}, which requests information for any child process in the same process group as the calling process,
+     *                  and any other negative value - pgid requests information for any child process whose process group ID is pgid.
+     * @param status    the status information from the child process is stored in this object, unless it's a null pointer.
+     * @param options   a bit mask. Its value should be the bitwise OR (that is, the ‘|’ operator) of zero or more of the WNOHANG and WUNTRACED flags.
+     *                  You can use the WNOHANG flag to indicate that the parent process shouldn't wait; and the WUNTRACED flag to request status information
+     *                  from stopped processes as well as processes that have terminated.
+     * @return the process ID of the child process whose status is reported otherwise -1 if it failed
+     * or errno ({@link ErrNo#EINTR}, {@link ErrNo#ECHILD}, {@link ErrNo#EINVAL}).
+     * If there are child processes but none of them is waiting to be noticed, waitpid will block until one is.
+     * However, if the WNOHANG option was specified, waitpid will return zero instead of blocking.
+     * If a specific PID to wait for was given, it will ignore all other children (if any). Therefore, if there are children waiting
+     * to be noticed but the child whose PID was specified is not one of them, waitpid will block or return zero as described above.
+     * @see <a href="https://www.freebsd.org/cgi/man.cgi?query=waitpid">waitpid(3)</a>
+     */
+    int waitpid(int pid, int status, int options);
```

Comment:
Changed statusPtr (type Object) to status (type int) because thats the actuall data type. But I'm not sure if JNA actually passes over the pointer of the java status int and lets its value be changeable in C.